js/dragonex.js

```diff
@@ -0,0 +1,912 @@
+        this.options['accessToken'] = token;
+        this.options['date'] = date;
+        return {};
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let request = '/' + this.implodeParams (path, params);
+        const query = this.omit (params, this.extractParams (path));
+        if (method === 'GET') {
+            if (query !== undefined) {
+                request += '?' + this.urlencode (query);
+            }
+        }
+        const url = this.urls['api'][api] + request;
+        if (api === 'private') {
+            body = JSON.stringify ({});
+            this.checkRequiredCredentials ();
+            if (method !== 'GET') {
+                if (query !== undefined) {
+                    body = this.json (query);
+                }
+            }
+            const contentMd5 = '';
+            const contentType = 'application/json';
+            const canonicalizedHeaders = '';
+            const ip = this.safeValue (params, 'bind_ip', '0.0.0.0');
+            const date = this.gmt (this.milliseconds ());
+            const strToTokenSign = ['POST', contentMd5, contentType, date, canonicalizedHeaders].join ('\n') + '/api/v1/token/new/';
+            let signature = this.hmac (this.encode (this.secret), this.encode (strToTokenSign), 'sha1', 'base64');
+            const authToken = this.apiKey + ':' + signature;
+            const tokenHeader = {
+                'Content-Sha1': contentMd5,
+                'Date': date,
+                'Content-Type': contentType,
+                'X-Real-IP-Proxy': ip,
+                'Auth': authToken,
+            };
+            const res = this.fetch ('https://openapi.dragonex.co/api/v1/token/new/', 'POST', tokenHeader);
+            const token = this.safeString (this.safeValue (res, 'data', {}), 'token', '');
+            let strToSign = [method.upper (), contentMd5, contentType, date, canonicalizedHeaders].join ('\n');
+            strToSign += request;
+            signature = this.hmac (this.encode (this.secret), this.encode (strToSign), 'sha1', 'base64');
+            const auth = this.apiKey + ':' + signature;
+            headers = {
+                'Content-Sha1': contentMd5,
+                'Date': date,
+                'Content-Type': contentType,
+                'X-Real-IP-Proxy': ip,
+                'Auth': auth,
+                'token': token,
+            };
+        }
+        return { 'url': url, 'method': method, 'body': body, 'headers': headers };
+    }
+
+    handleErrors (code, reason, url, method, headers, body, response, requestHeaders, requestBody) {
+        if (response === undefined) {
+            return; // fallback to default error handler
+        }
+        if ('code' in response) {
+            if (response['code'] !== 1) {
+                throw new ExchangeError (this.id + ' ' + JSON.parse (response));
```

Comment:
We need to add the proper exception handing, see how `throwExactlyMatchedException` and `throwBroadlyMatchedException` is called in the error handlers of the other exchanges. The error maps goes to `this.exceptions` property.